test/unit/test_require_interface.py

```diff
@@ -36,10 +40,10 @@ def test_require():
         "access-key": "my-access-key",
         "secret-key": "my-secret-key",
     }
-    harness.update_relation_data(rel_id, "test-app2", {"data": yaml.dump(data)})
+    harness.update_relation_data(rel_id, "test-app2", {"data-v1": yaml.dump(data)})
     assert harness.charm.interface.is_available
 
-    rel_data = harness.charm.interface.data[0]
+    rel_data = harness.charm.interface.get_data()[0]
     assert rel_data == data
 
```

Comment:
Maybe another test case which covers both Charms sending data would make sense.

Reply:
Good call, updated to include that